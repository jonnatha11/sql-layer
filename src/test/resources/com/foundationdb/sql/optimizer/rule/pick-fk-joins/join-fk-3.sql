select FATHER.name, MOTHER.name, animal.description
from parent AS FATHER, parent as MOTHER, animal
where animal.id = 42
AND FATHER.id = animal.father_id
AND animal.mother_id = MOTHER.id