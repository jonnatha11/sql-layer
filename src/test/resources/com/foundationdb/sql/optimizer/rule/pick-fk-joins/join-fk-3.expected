SelectQuery@2399a19d
  ResultSet@3b415cad[name, name, description]
    Project@402b803f[father.name, mother.name, animal.description]
      Select@67f0c902[]
        JoinNode@4d654e9e(INNER/NESTED_LOOPS[mother.id == animal.mother_id])
          TableGroupJoinTree@76e1f7bf(TableGroup@2b4b2dd(parent), mother - SingleIndexScan@360c8ec4(Index(test.parent.name[IndexColumn(name)]), covering/NONE, rows = 1, cost = 7.17100))
            TableSource@4c98bd99(mother - TableGroup@2b4b2dd(parent))
          JoinNode@3281dca5(INNER/NESTED_LOOPS[father.id == animal.father_id])
            TableGroupJoinTree@5827384f(TableGroup@48a9c8a8(parent), father - SingleIndexScan@1d7d593b(Index(test.parent.name[IndexColumn(name)]), covering/NONE, rows = 1, cost = 7.17100))
              TableSource@71a80002(father - TableGroup@48a9c8a8(parent))
            TableGroupJoinTree@986fff7(TableGroup@254acc8e(animal), animal - ExpressionsHKeyScan@6bbcee8a(TableSource@398f573b(animal - TableGroup@254acc8e(animal)), 42, rows = 1, cost = 10.7102))
              TableSource@398f573b(animal - TableGroup@254acc8e(animal))
